src/broyden.jl

```diff
@@ -0,0 +1,56 @@
+"""
+```julia
+Broyden()
+```
+
+A low-overhead implementation of Broyden. This method is non-allocating on scalar
+and static array problems.
+"""
+struct Broyden <: AbstractSimpleNonlinearSolveAlgorithm end
+
+function SciMLBase.solve(prob::NonlinearProblem,
+                         alg::Broyden, args...; abstol = nothing,
+                         reltol = nothing,
+                         maxiters = 1000, kwargs...)
+    f = Base.Fix2(prob.f, prob.p)
+    x = float(prob.u0)
+    fₙ = f(x)
+    T = eltype(x)
+    if length(x) > 1
+        J⁻¹ = Matrix{T}(I, length(x), length(x))
+    else
+        J⁻¹ = 1
+    end
```

Comment:
```suggestion
    J⁻¹ = ArrayInterfaceCore.zeromatrix(x)
```

This should make it GPU and static array safe.